### ISSUE TITLE
Error in normalizePath on Windows with Unicode path characters (https://github.com/rstudio/rmarkdown/issues/1529)

### ISSUE THREAD

**Issue by fotisfotiadis** [rstudio/rmarkdown#2127]
Hi all
I am having an issue with publishing an R markdown document,
When trying to knit an rmd file, that's in a path with greek characters, I get this error:

Error in file(con, "r") : cannot open the connection
In addition: Warning message:
In file(con, "r") :
  cannot open file 'C:/Users/tsarl/Desktop/Ξ•Ξ”Ξ™Ξ /Paper/R/Delete.Rmd': No such file or directory

The html file is well created, and in the R markdown console there is this message:
rmarkdown::render('C:/Users/tsarl/Desktop/\u{395}\u{394}\u{399}\u{3a0}/Paper/R/Delete.Rmd',  encoding = 'UTF-8')

So, this is obviously an encoding thing, but there seems to be no solution, since the bug is considered to be fixed: https://github.com/rstudio/rmarkdown/issues/1529 

I updated all the packages, and restarted R studio several times. No luck.
Is it a Windows 7 thing perhaps (yes, I am still using Win 7)? I include my session info:

R version 4.0.4 (2021-02-15)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=Greek_Greece.1253  LC_CTYPE=Greek_Greece.1253    LC_MONETARY=Greek_Greece.1253
[4] LC_NUMERIC=C                  LC_TIME=Greek_Greece.1253    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] compiler_4.0.4    rsconnect_0.8.17  htmltools_0.5.1.1 tools_4.0.4       yaml_2.2.1        tinytex_0.31     
 [7] rmarkdown_2.7     knitr_1.33        digest_0.6.27     xfun_0.22         rlang_0.4.10      evaluate_0.14 

Thank you so much for your time. 
Fotis

**Comment by cderv** [comment]
Hi @fotisfotiadis, thanks for the report. 

Would you be able to share the traceback of the R error you get ? 
* https://rstats.wtf/debugging-r-code.html#debugging-your-own-code
* https://adv-r.hadley.nz/debugging.html?q=traceback#debugging
This would help us know which function is impacted by this. 

I suspect this could be still be related to the issue you linked to and we may have missed something. 

Thank you.

**Comment by fotisfotiadis** [comment]
Hi @cderv 
Thank you for your response.
The path in question is C:/Users/tsarl/Desktop/ΕΔΙΠ/Paper/R

And the traceback() functions outputs this:
```
Error in file(con, "r") : cannot open the connection
In addition: Warning message:
In file(con, "r") :
  cannot open file 'C:/Users/tsarl/Desktop/Ξ•Ξ”Ξ™Ξ /Paper/R/Test.Rmd': No such file or directory
> traceback()
3: file(con, "r")
2: readLines(target, encoding = encoding, warn = FALSE)
1: (function (target, encoding) 
   {
       lines <- readLines(target, encoding = encoding, warn = FALSE)
       frontMatter <- rmarkdown:::parse_yaml_front_matter(lines)
       if (is.list(frontMatter) && (identical(frontMatter$runtime, 
           "shiny_prerendered") || identical(frontMatter$runtime, 
           "shinyrmd"))) {
           is_multi_rmd <- FALSE
       }
       else {
           rmds <- list.files(path = dirname(target), pattern = glob2rx("*.Rmd"), 
               all.files = FALSE, recursive = FALSE, ignore.case = TRUE, 
               include.dirs = FALSE)
           is_multi_rmd <- length(rmds) > 1
       }
       selfContained <- TRUE
       outputFormat <- rmarkdown:::output_format_from_yaml_front_matter(lines)
       if (is.list(outputFormat$options) && identical(outputFormat$options$self_contained, 
           FALSE)) {
           selfContained <- FALSE
       }
       title <- ""
       if (is.list(frontMatter) && is.character(frontMatter$title)) {
           title <- frontMatter$title
       }
       renderFunction <- .rs.getCustomRenderFunction(target)
       list(is_multi_rmd = .rs.scalar(is_multi_rmd), is_shiny_rmd = .rs.scalar(renderFunction == 
           "rmarkdown::run"), is_self_contained = .rs.scalar(selfContained), 
           title = .rs.scalar(title), has_connect_account = .rs.scalar(.rs.hasConnectAccount()))
   })("C:/Users/tsarl/Desktop/Ξ•Ξ”Ξ\231Ξ /Paper/R/Test.Rmd", "UTF-8")
```

Hope it's more clear now

Regards,
Fotis

**Comment by cderv** [comment]
Oh interesting ! This is a function from the RStudio IDE 

Two more things: 

* Which version of RStudio are you using ? 
* Can you render using th R Console directly with `rmarkdown::render()` ? And if error, share the traceback too.

Thanks

**Comment by fotisfotiadis** [comment]
I am using R Studio Version 1.4.1106 

The markdown::render did not produce any errors. The html file was created:

```
rmarkdown::render('Test.Rmd')


processing file: Test.Rmd
  |................                                                                                               |  14%
  ordinary text without R code

  |................................                                                                               |  29%
label: setup (with options) 
List of 1
 $ include: logi FALSE

  |................................................                                                               |  43%
  ordinary text without R code

  |...............................................................                                                |  57%
label: cars
  |...............................................................................                                |  71%
  ordinary text without R code

  |...............................................................................................                |  86%
label: pressure (with options) 
List of 1
 $ echo: logi FALSE

  |...............................................................................................................| 100%
  ordinary text without R code


output file: Test.knit.md

"C:/Program Files/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS Test.utf8.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output Test.html --lua-filter "C:\Users\tsarl\Documents\R\win-library\4.0\rmarkdown\rmarkdown\lua\pagebreak.lua" --lua-filter "C:\Users\tsarl\Documents\R\win-library\4.0\rmarkdown\rmarkdown\lua\latex-div.lua" --self-contained --variable bs3=TRUE --standalone --section-divs --template "C:\Users\tsarl\Documents\R\win-library\4.0\rmarkdown\rmd\h\default.html" --no-highlight --variable highlightjs=1 --variable theme=bootstrap --include-in-header "C:\Users\tsarl\AppData\Local\Temp\RtmpYvu7ih\rmarkdown-str3a602a755f8f.html" --mathjax --variable "mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" 

Output created: Test.html
```

Interestingly, in the R markdown tab, there is this line of code:
`rmarkdown::render('C:/Users/tsarl/Desktop/\u{395}\u{394}\u{399}\u{3a0}/Paper/R/Test.Rmd',  encoding = 'UTF-8');`


Regards,
Fotis

**Comment by cderv** [comment]
> The markdown::render did not produce any errors. The html file was created:

Can you try also by rendering with the full path ? 
Does not `normalizePath("Test.Rmd")` output a path with correct characters ? 

> Interestingly, in the R markdown tab, there is this line of code:
rmarkdown::render('C:/Users/tsarl/Desktop/\u{395}\u{394}\u{399}\u{3a0}/Paper/R/Test.Rmd', encoding = 'UTF-8');

This command line is created by RStudio IDE, so it could be an issue with the IDE not correctly retrieving the path of the file in the correct encoding. 

We may need to move this issue there

**Comment by fotisfotiadis** [comment]
Hallo

Rendering twith the full path again produced to error
```

rmarkdown::render('C:/Users/tsarl/Desktop/ΕΔΙΠ/Paper/R/Test.Rmd');


processing file: Test.Rmd
  |................                                                                                               |  14%
  ordinary text without R code

  |................................                                                                               |  29%
label: setup (with options) 
List of 1
 $ include: logi FALSE

  |................................................                                                               |  43%
  ordinary text without R code

  |...............................................................                                                |  57%
label: cars
  |...............................................................................                                |  71%
  ordinary text without R code

  |...............................................................................................                |  86%
label: pressure (with options) 
List of 1
 $ echo: logi FALSE

  |...............................................................................................................| 100%
  ordinary text without R code


output file: Test.knit.md

"C:/Program Files/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS Test.utf8.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output Test.html --lua-filter "C:\Users\tsarl\Documents\R\win-library\4.0\rmarkdown\rmarkdown\lua\pagebreak.lua" --lua-filter "C:\Users\tsarl\Documents\R\win-library\4.0\rmarkdown\rmarkdown\lua\latex-div.lua" --self-contained --variable bs3=TRUE --standalone --section-divs --template "C:\Users\tsarl\Documents\R\win-library\4.0\rmarkdown\rmd\h\default.html" --no-highlight --variable highlightjs=1 --variable theme=bootstrap --include-in-header "C:\Users\tsarl\AppData\Local\Temp\RtmpSGxrqp\rmarkdown-str1fe07b0f19fe.html" --mathjax --variable "mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" 

Output created: Test.html
```

And Normalize Path gives this output

```
> normalizePath("Test.Rmd")
[1] "C:\\Users\\tsarl\\Desktop\\ΕΔΙΠ\\Paper\\R\\Test.Rmd"
```

Is it more clear now?
Regards,
Fotis

**Comment by cderv** [comment]
> Rendering twith the full path again produced to error

Do you mean produced **no** error ? The code you pasted seems to be working. 

This new information makes me thing this is something that as to do with how the RStudio IDE handles file path.

Maybe this is fixed in recent version. You could try the last daily version if you are willing to. There is a installer-less version that should not impact your current installation: https://dailies.rstudio.com/rstudio/oss/windows/xcopy/

If this still happens then I would suggest opening an issue in RStudio Github repo.

Pinging @kevinushey in case he has some thoughts on this UTF-8 path encoding thing.

**Comment by kevinushey** [comment]
The bug is coming from RStudio here:

https://github.com/rstudio/rstudio/blob/147cc86cec36bae1536e3fe56a078ae618bcb445/src/cpp/session/modules/SessionRSConnect.R#L464-L507

We probably need to mark the encoding of the 'target' file path as UTF-8.

**Comment by kevinushey** [comment]
Just to confirm:

```
> Sys.setlocale(locale = "Greek")
[1] "LC_COLLATE=Greek_Greece.1253;LC_CTYPE=Greek_Greece.1253;LC_MONETARY=Greek_Greece.1253;LC_NUMERIC=C;LC_TIME=Greek_Greece.1253"
> 
> path <- 'C:/Users/tsarl/Desktop/\u{395}\u{394}\u{399}\u{3a0}/Paper/R/Delete.Rmd'
> path
[1] "C:/Users/tsarl/Desktop/ΕΔΙΠ/Paper/R/Delete.Rmd"
> Encoding(path) <- "native"
> path
[1] "C:/Users/tsarl/Desktop/Ξ•Ξ”Ξ™Ξ /Paper/R/Delete.Rmd"
```

so we're accidentally mis-encoding a UTF-8 path as though it were native (ie, current code page; on your system Greek)

**Comment by kevinushey** [comment]
It would be worth testing whether you still see this issue in the latest daily builds: https://dailies.rstudio.com
